vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/application_code/nuvoton_code/entropy_hardware_poll.c

```diff
@@ -0,0 +1,221 @@
+/*
+ *  Hardware entropy collector for M487 RNG
+ *
+ */
+
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include "FreeRTOS.h"
+#include "task.h"
+#include "NuMicro.h"
+
+/*
+ * Get Random number generator.
+ */
+ #define PRNG_KEY_SIZE  (0x20UL)
+ 
+static volatile int  g_PRNG_done;
+volatile int  g_AES_done;
+
+
+#define SNUM        32       /* recorded number of lastest samples */
+
+
+static uint32_t   adc_val[SNUM];
```

Comment:
Why are the above globals declared volatile but not these?

Reply:
@dcgaws , to avoid compiler optimization, it will be utilized in TLS with future HW AES acceleration.

Reply:
I don't understand. Would a more detailed explanation be available?

Reply:
@dcgaws , in the future, we could implement AES HW acceleration in mbed-tls. AES-done flag will be set in xxx_pool.c and read in mbed-tls aes part. Set volatile to avoid compiler optimization.